docs/tools/illink/data-formats.md

```diff
@@ -1,5 +1,9 @@
 # Data Formats
 
+> [!WARNING]
+> The file formats described in this file may not be supported for use outside the dotnet/runtime repo. Refer to the official documentation at https://learn.microsoft.com/dotnet/core/deploying/trimming/trim-self-contained for supported formats.
```

Comment:
https://learn.microsoft.com/en-gb/dotnet/core/deploying/trimming/trimming-options#root-descriptors links back to this document. This means that only root descriptors are supported for external use?

Reply:
> https://learn.microsoft.com/en-gb/dotnet/core/deploying/trimming/trimming-options#root-descriptors links back to this document. This means that only root descriptors are supported for external use?

We very likely do not want to link to this doc from learn, whatever is supported should be moved to the docs repo. I guess that link explains how a customer dug out the attribute XML format the other day (cc @sbomer @agocke).

Reply:
Are we sure the rest is supported?

I just quickly skimmed the doc - how about the "Override static field value with a constant" part for example?

Reply:
> I think we weren't very deliberate about defining the support level of these features when they were introduced.

This is a general problem with having official documentation in the dotnet/runtime repo. These files are not thought about as public contracts - neither the author of the doc nor the reviewers look at it that way.

Official doc changes are always looked at as public contracts because there's a clear context switch between "I'm developing a feature" and "I'm documenting a contract". Plus doing that in the docs repo ensures the doc matches the standards of official docs by getting a review from relevant doc subject matter experts.

Reply:
In an unreleased library of mine, I am using a substitutions file to trim the resource keys if resource keys are enabled, copying the BCL's mechanism. Also F# is using a substitutions file to remove some big binary resources from `FSharp.Core` with signature data (which I copied to a released F# library of mine).

Are these not supported for external use?

Reply:
Do we have a recommendation for what these places should use instead?